recipes/install.rb

```diff
@@ -70,11 +72,12 @@
 end
 
 # Create keyFile if specified
-file node['mongodb']['config']['keyFile'] do
-  owner node['mongodb']['user']
-  group node['mongodb']['group']
-  mode  '0600'
-  backup false
-  content node['mongodb']['key_file_content']
-  only_if { node['mongodb']['key_file_content'] }
+if node['mongodb']['key_file_content']
+  file node['mongodb']['config']['keyFile'] do
+    owner node['mongodb']['user']
+    group node['mongodb']['group']
+    mode  '0600'
+    backup false
+    content node['mongodb']['key_file_content']
+  end
```

Comment:
I suppose there will be a good reason. So I just ask from my ignorance: why have you replaced the `only_if` here with an external `if`?